docs/src/main/asciidoc/spanner.adoc

```diff
@@ -0,0 +1,149 @@
+However, Hibernate-side cascading operations such as `@ManyToOne(cascade = {CascadeType.ALL})` are supported.
+
+
+==== Constraints
+
+Cloud Spanner does not support constraints.
+As a result, `SpannerDialect` does not currently support any constraints such as `FOREIGN KEY`, `UNIQUE`, or `ON DELETE CASCADE`.
+
+NOTE: Hibernate doesn't directly rely on the existence of constraints to perform its operations and leaves the enforcement relationship links to the database.
+
+
+==== Collections
+
+`SpannerDialect` supports collection properties annotated with `@ElementCollection`.
+However, as with inter-object-relationships, foreign-key-columns in collection tables do not have the Foreign Key constraint.
+
+==== Table Catalogs and Schemas
+
+Cloud Spanner does not support table names with Catalog and Schema components:
+
+[source, java]
+----
+// Not supported: `public.store.book` is not a valid Cloud Spanner table name.
+@Table(
+  catalog = "public",
+  schema = "store",
+  name = "book"
+)
+
+// Supported.
+@Table(
+  name = "book"
+)
+----
+
+
+==== @ColumnDefault
+
+The dialect does not currently set default values based on the https://docs.jboss.org/hibernate/orm/5.4/javadocs/org/hibernate/annotations/ColumnDefault.html[`@ColumnDefault`] annotation,
+because `null` values aren't specially handled and are stored just like other values by Cloud Spanner and its driver.
+
+
+==== Decimal and Numeric Types
+
+Cloud Spanner does not provide native support for https://cloud.google.com/spanner/docs/storing-numeric-data[arbitrary-precision decimal numbers], but does provide some alternative approaches.
+As a result, the driver and dialect do not support decimal and arbitrary-precision Java types that Hibernate maps to Java Persistence types `java.sql.Type` of `NUMERIC` and `DECIMAL` such as `java.math.BigDecimal`.
+
+
+==== Subclasses using `InheritanceType.JOINED`
+
+If you are using entities that are related by inheritance with the `@Inheritance(strategy = InheritanceType.JOINED)`:
+
+[source, java]
+----
+@Entity
+@Inheritance(strategy = InheritanceType.JOINED)
+public class Payment {
+
+    @Id
+    @GeneratedValue
+    private Long id;
+
+    private Long amount;
+}
+
+@Entity
+public class WireTransferPayment extends Payment {
+}
+
+@Entity
+public class CreditCardPayment extends Payment {
+}
+----
+
+You must set the `hibernate.hql.bulk_id_strategy` setting in `hibernate.properties` to one of the following classes:
+
+- `InlineIdsInClauseBulkIdStrategy`
+- `InlineIdsSubSelectValueListBulkIdStrategy`
+- `InlineIdsOrClauseBulkIdStrategy`
+- `CteValuesListBulkIdStrategy`
+
+This is because Hibernate's default behavior attempts to create intermediate tables to handle delete and update operations on the multiple tables that represent a `JOINED` inheritance hierarchy, but these table creations statements do not conform to Cloud Spanner DDL.
+Using one of the `Inline` bulk-ID strategy classes given above resolves this issue.
+
+
```

Comment:
I think the section about UUID generation and caveats around that is missing.

Reply:
@meltsufin added a new section and table as requested, PTAL